### DISCUSSION TITLE
[Chapter 10] Redis와 메시지 브로커

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#165] [Chapter 10. Redis and Spring Cache]
```
10.7 부분을 보면 레디스가 제공하는 Pub-Sub 은 Kafka 혹은 RabbitMQ 같은 메시지 큐에 비해 비교적 간단하고 빠른 기능을 제공하나 
이들과 다른점은 구독자가 메시지 수신에 실패해도 레디스는 실패한 메시지를 전달할 수 없다.
```

 는 문구가 있습니다.  
 
 메시지 수신을 실패해도 다시 전달할 수 없다? 메시지 브로커를 사용하는 이유 중 저는 개인적으로 보장성을 지킬 수 있는 점이 매우 크다고 생각하는데, 실패한 메시지를 다시 전달할 수 없다면 보장성을 지킬 수 없고 이렇다면 왜 레디스를 메시지 브로커로써 사용하는지 궁금합니다.  

어떤 경우에 레디스를 이용한 Pub-Sub을 구현하여 사용하는지 실무에서 사용하시거나, 다른 사용 예시를 아신다면 공유해주시면 감사하겠습니다.

**Comment by JoisFe** [comment]
이걸 보면 TCP 와 UDP 통신이 생각이 나네요
UDP 통신 경우 신뢰성 있게 전송하지 않기 때문에, 데이터 손실이나 순서 변경이 발생할 수 있지만 영상 스트리밍 처럼 간단하고 빠르게 데이터 통신이 이루어져야 하는 경우 사용한다고 합니다.

이와 비슷하게 간단하고 빠르게 메시지 송수신을 해야하는 경우 RabbitMQ, Kafka 대신 레디스를 사용하지 않을까? 생각 합니다.

**Comment by corock** [comment]
레디스는 사용 용도가 다양해서, 메시지 브로커를 소개하는 건 이런 용도로도 쓸 수 있다~ 정도가 아닐까요?

**Comment by JoisFe** [comment]
https://redis.com/solutions/use-cases/messaging/
